airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/pipeline.py

```diff
@@ -39,27 +55,105 @@ def get_test_steps(context: ConnectorContext) -> STEP_TREE:
+            raise ValueError(
+                f"Invalid step name {step_name}, it must be one of {[step_id.value for step_id in unique_steps_to_run.keys()]}"
+            )
+        step_id = CONNECTOR_TEST_STEP_ID(step_name)
+        if step_id not in extra_params_mapping:
+            extra_params_mapping[step_id] = {}
+        if param_name not in extra_params_mapping[step_id]:
+            extra_params_mapping[step_id][param_name] = []
+        if param_value is not None:
+            extra_params_mapping[step_id][param_name].append(param_value)
+    return extra_params_mapping
+
+
+def log_steps_params(unique_steps_to_run: Dict[CONNECTOR_TEST_STEP_ID, Step]) -> None:
+    """Log the steps params.
+
+    Args:
+        unique_steps_to_run (Dict[CONNECTOR_TEST_STEP_ID, Step]): The unique steps to run in the test pipeline.
+    """
+    for step_id, step in unique_steps_to_run.items():
+        if step.params:
+            step.logger.info(f"Step ID {step_id} will run with the following parameters: {step.params}")
+
+
+def set_extra_params_on_steps_to_run(extra_params: Tuple[str], steps_to_run: STEP_TREE) -> None:
+    """Set the extra params on the steps to run.
+
+    Args:
+        extra_params (Tuple[str]): The extra params provided by the user.
+        steps_to_run (STEP_TREE): The steps to run in the test pipeline.
+    """
+    # As the steps to run can be nested, we need to flatten them to get the unique steps to run
+    unique_steps_to_run = {step_to_run.id: step_to_run.step for step_to_run in collapse(steps_to_run)}
+    extra_params_mapping = build_extra_params_mapping(extra_params, unique_steps_to_run)
+    for step_id, validated_extra_params in extra_params_mapping.items():
+        unique_steps_to_run[step_id].extra_params = validated_extra_params
+    log_steps_params(unique_steps_to_run)
+
+
+async def run_connector_test_pipeline(context: ConnectorContext, semaphore: anyio.Semaphore, extra_params: Tuple[str]) -> ConnectorReport:
     """
     Compute the steps to run for a connector test pipeline.
     """
+    all_steps_to_run: STEP_TREE = []
 
-    steps_to_run = get_test_steps(context)
+    all_steps_to_run += get_test_steps(context)
 
     if not context.code_tests_only:
-        steps_to_run += [
+        static_analysis_steps_to_run = [
             [
                 StepToRun(id=CONNECTOR_TEST_STEP_ID.METADATA_VALIDATION, step=MetadataValidation(context)),
                 StepToRun(id=CONNECTOR_TEST_STEP_ID.VERSION_FOLLOW_CHECK, step=VersionFollowsSemverCheck(context)),
                 StepToRun(id=CONNECTOR_TEST_STEP_ID.VERSION_INC_CHECK, step=VersionIncrementCheck(context)),
                 StepToRun(id=CONNECTOR_TEST_STEP_ID.QA_CHECKS, step=QaChecks(context)),
             ]
         ]
+        all_steps_to_run += static_analysis_steps_to_run
+
+    set_extra_params_on_steps_to_run(extra_params, all_steps_to_run)
```

Comment:
❗This method is best placed inside `run_steps`. As this is the area responsible for evaluating the step tree at run time, and extra_args are a runtime concern.